tests/dragonfly/replication_test.py

```diff
@@ -2568,4 +2572,53 @@ async def check_flag(key, flag):
     await check_flag("key1", 2)
     await check_flag("key2", 123456)
 
-    await close_clients(c_replica)
+    for i in range(1, 100):
+        assert c_mc_replica.get(f"key{i}") == str.encode(f"value{i}")
+
+    await close_clients(c_replica, c_master)
+
+
+@pytest.mark.parametrize("master_threads, replica_threads", [[4, 4]])
+@pytest.mark.asyncio
+async def test_double_take_over(df_factory, df_seeder_factory, master_threads, replica_threads):
+    master = df_factory.create(
+        proactor_threads=master_threads, dbfilename="", admin_port=ADMIN_PORT
+    )
+    replica = df_factory.create(
+        proactor_threads=replica_threads, dbfilename="", admin_port=ADMIN_PORT + 1
+    )
+    df_factory.start_all([master, replica])
+
+    seeder = df_seeder_factory.create(port=master.port, keys=1000, dbcount=5, stop_on_failure=False)
+    await seeder.run(target_deviation=0.1)
+
+    capture = await seeder.capture(port=master.port)
+
+    c_replica = replica.client()
+
+    logging.debug("start replication")
+    await c_replica.execute_command(f"REPLICAOF localhost {master.admin_port}")
+    await wait_available_async(c_replica)
+
+    logging.debug("running repltakover")
+    await c_replica.execute_command(f"REPLTAKEOVER 10")
+    assert await c_replica.execute_command("role") == ["master", []]
+
+    await asyncio.sleep(0.1)
+    assert master.proc.poll() == 0, "Master process did not exit correctly."
```

Comment:
Please use `assert_eventually` here, this can definitely take longer than 0.1 seconds